pyproject.toml

```diff
@@ -63,6 +65,8 @@ docs = [
     "sphinxext-opengraph",
     "watermark",
     "mlflow>=2.0.0",
+    "networkx",
+    "dowhy",
```

Comment:
I think this should not be core dependencies but rather optional (like numpyro for sampling). See the comment above on abstraction.

Reply:
I'm following @wd60622 idea and moving this to optional dependencies.